src/data/quot.lean

```diff
@@ -551,4 +585,38 @@ protected lemma mk'_eq_mk (x : α) : quotient.mk' x = ⟦x⟧ := rfl
 
 end
 
+/-! #### Decidability -/
+
+instance (f : α → Prop) (h : ∀ a b, @setoid.r α s₁ a b → f a = f b) [decidable_pred f] :
+  decidable_pred (quotient.lift f h) :=
+quot.lift.decidable_pred _ _ _
+
+instance (f : α → β → Prop)
+  (h : ∀ a₁ a₂ b₁ b₂, @setoid.r α s₁ a₁ b₁ → @setoid.r β s₂ a₂ b₂ → f a₁ a₂ = f b₁ b₂)
+  [hf : Π a, decidable_pred (f a)] (q₁ : quotient s₁) :
+  decidable_pred (quotient.lift₂ f h q₁) :=
+λ q₂, quotient.rec_on_subsingleton₂' q₁ q₂ hf
+
+instance (q : quotient s₁) (f : α → Prop) (h : ∀ a b, @setoid.r α s₁ a b → f a = f b)
+  [decidable_pred f] :
+  decidable (quotient.lift_on q f h) :=
+quotient.lift.decidable_pred _ _ _
+
+instance (q₁ : quotient s₁) (q₂ : quotient s₂) (f : α → β → Prop)
+  (h : ∀ a₁ a₂ b₁ b₂, @setoid.r α s₁ a₁ b₁ → @setoid.r β s₂ a₂ b₂ → f a₁ a₂ = f b₁ b₂)
```

Comment:
Shouldn't `h` be stated with `has_equiv.equiv` to match the type of `quotient.lift_on₂`?

Reply:
It makes it more annoying to apply. See #11824

Reply:
Right, but you should never have to apply it when the typeclass isn't available, because you should be using the primed version instead in those cases.

Reply:
I guess we should get rid of the unprimed version altogether...